testing/test_warnings.py

```diff
@@ -641,3 +642,151 @@ def pytest_configure():
     assert "INTERNALERROR" not in result.stderr.str()
     warning = recwarn.pop()
     assert str(warning.message) == "from pytest_configure"
+
+
+class TestStackLevel:
+    @pytest.fixture
+    def capwarn(self):
+        class CapturedWarnings:
+            captured = None
+
+            @classmethod
+            def pytest_warning_captured(cls, warning_message, when, item, context):
+                cls.captured = (warning_message, context)
+
+        return CapturedWarnings
+
+    def test_issue4445_rewrite(self, testdir, capwarn):
+        """#4445: Make sure the warning points to a reasonable location
+        See origin of _issue_warning_captured at: _pytest.assertion.rewrite.py:241
+        """
+        testdir.plugins = [capwarn()]
+        testdir.makepyfile(some_mod="")
+        conftest = testdir.makeconftest(
+            """
+                import some_mod
+                import pytest
+
+                pytest.register_assert_rewrite("some_mod")
+            """
+        )
+        testdir.parseconfig()
+
+        # with stacklevel=5 the warning originates from register_assert_rewrite
+        # function in the created conftest.py
+        warning, context = capwarn.captured
```

Comment:
Following the suggestion of `capwarn` returning a list, this would then be:

```python
assert len(capwarn) == 1
warning, context = capwarn[0]
```

Reply:
I find this another good step in making the tests stronger! 

However i found out strange things while enhancing the tests with your suggestions regarding using `len(capwarn)`. Depending on the order of calls it can happen, that some of the warning emitting parts are called twice (like cacheprovider.set and import_plugin) so that two identical warnings are captured. I only know few to nothing about the respective parts that account for the call order but for me it is worth to mention that. I will dig deeper with the debugger and enhance the respective tests with comments about my findings. 

But maybe this has to do with how pytester works and i am lacking necessary understanding of that procedure ;)